File: app/src/main/java/id/ac/stiki/doleno/absenin/data/source/network/AbsentStore.java
package id.ac.stiki.doleno.absenin.data.source.network;

import com.google.android.gms.tasks.Task;
import com.google.firebase.firestore.QuerySnapshot;

import id.ac.stiki.doleno.absenin.data.entity.Absent;

public interface AbsentStore {
    Task<Void> createAbsent(Absent event, String email);

    Task<Void> updateAbsent(Absent event, String email);

    Task<QuerySnapshot> get(String email);
}

File: app/src/main/java/id/ac/stiki/doleno/absenin/data/source/network/AbsentStore.kt
package id.ac.stiki.doleno.absenin.data.source.network

import com.google.android.gms.tasks.Task
import com.google.firebase.firestore.QuerySnapshot
import id.ac.stiki.doleno.absenin.data.entity.Absent

interface AbsentStore {
    fun createAbsent(absent: Absent, email: String): Task<Void>
    fun updateAbsent(absent: Absent, email: String): Task<Void>
    operator fun get(email: String): Task<QuerySnapshot>
    operator fun get(id: Long, email: String): Absent
}